File: PortalDietetycznyAPI/Domain/DTOs/AddRecipeDto.cs
using PortalDietetycznyAPI.Domain.Entities;

namespace PortalDietetycznyAPI.DTOs;

public class AddRecipeDto
{
    public List<int> TagsIds { get; set; }
    public string Name { get; set; }
    public List<RecipeIngredientDto> Ingredients { get; set; }
    public NutritionInfo NutritionInfo { get; set; } 
    public string Instruction { get; set; }
    public string FileBytes { get; set; }
    public string FileName { get; set; }
}

File: PortalDietetyczny.Client/src/app/DTOs/AddRecipeDto.ts
export interface AddRecipeDto
{
  tagsIds: number[];
  name: string;
  ingredients: RecipeIngredientDto[];
  nutritionInfo: NutritionInfo;
  instruction: string;
  fileBytes: string
  fileName: string
}
export interface RecipeIngredientDto {
  id: number;
  name: string;
  unit: string;
  unitValue: number;
  homeUnit: string;
  homeUnitValue: number;
}

export interface NutritionInfo {
  fiber: number;
  fat: number;
  carb: number;
  protein: number;
  kcal: number;
}
